crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -265,6 +281,24 @@ impl CheckpointSummary {
     pub fn is_last_checkpoint_of_epoch(&self) -> bool {
         self.end_of_epoch_data.is_some()
     }
+
+    pub fn version_specific_data(
+        &self,
+        config: &ProtocolConfig,
+    ) -> Option<CheckpointVersionSpecificData> {
+        match config.checkpoint_summary_version_specific_data_as_option() {
+            None | Some(0) => None,
+            Some(1) => Some(
+                bcs::from_bytes(&self.version_specific_data).unwrap_or_else(|e| {
+                    panic!(
```

Comment:
since this is not on a CertifiedCheckpointSummary, we should probably return an error instead of panicking. Caller can panic if the summary was certified (and verified!)

Reply:
done